### PR TITLE
Use specific version of yapf to avoid format diffs

### PR DESCRIPTION
**Pull request by stevemessick**
Switch to Google's open-source project formatting conventions, `yapf`, except preserve our 4-space indentation.

See #635 for results. If we need to change things even more, now is a good time for feedback.

### CODE REVIEW

File: pyproject.toml
Comment (jmasukawa) [inline pyproject.toml:61]: re: `--break-system-packages` i think *we're* ok with this ("we" as in kaggle devs), but i could see someone else wanting to contribute to this project and not expecting to have this behavior.

we may want to look at setting up a python environment, or perhaps even better, a dev docker container for the Kaggle API so this would be contained to that. (and if we already have that, apologies for the noise)

Comment (jmasukawa) [inline pyproject.toml:61]: IIRC when i was looking at this, some environments people worked from (cloudtop vs. `kaggle-web-dev`) may not have had the same `yapf` package availability. hoping this works on both now? 🥺

**Review by jmasukawa — APPROVED** [review]
intent LGTM, approving to unblock

File: tools/GeneratePythonLibrary.sh
Comment (jmasukawa) [inline tools/GeneratePythonLibrary.sh:93]: tbh i kind of like google's 2-space indentation since it's more compact, but no strong feelings, open to whatever the group prefers.

https://engdoc.corp.google.com/eng/doc/devguide/py/style/index.md?cl=head#s3.4-indentation

File: pyproject.toml
Comment (stevemessick) [reply pyproject.toml:61]: Ah, good point. I need to look into this further.

There is a Docker container that's used for distribution. I will see if I can launch it locally and do some testing.

File: tools/GeneratePythonLibrary.sh
Comment (stevemessick) [reply tools/GeneratePythonLibrary.sh:93]: I sorta prefer two as well, and that's what's used in `kapigen`. It's easier to see the non-indent changes this way. If others agree then I think we can change to two before I merge this.

File: pyproject.toml
Comment (jplotts) [reply pyproject.toml:61]: Agreed, this (creating a dev Docker container) is a long-standing request.  I'm not sure if it needs to be done immediately, but it seems to just keep biting us.

File: tools/GeneratePythonLibrary.sh
Comment (jplotts) [reply tools/GeneratePythonLibrary.sh:93]: Agreed on preferring compact.  Can you change it and then update PR #635 ?

**Review by jplotts — APPROVED** [review]
LGTM pending 2-space formatting change.

**Review by rosbo — APPROVED** [review]

**Pull request merged**
